server/methods/core/cart.js

```diff
@@ -1150,6 +1150,9 @@ Meteor.methods({
       });
     }
 
+    // e.g. discount records would be already present on the billing array. Add to the end of the array.
+    Array.prototype.push.apply(payments, cart.billing.filter((billingInfo) => billingInfo.paymentMethod));
```

Comment:
This is clever but also a little difficult to read.
What are we filtering here, billingInfo that has paymentMethod?

Would this be identical in function?
```js
const paymentMethods = cart.billing.filter((billingInfo) => billingInfo.paymentMethod));
payments.push(...paymentMethods)
```

Looking at it further, it seems that payments is always initialized to an empty array. What is the difference between `payments` and the result of the `cart.billing.filter` function?

e.g.
```
const payments = cart.billing.filter((billingInfo) => billingInfo.paymentMethod));
```

Reply:
Agree, the spread operator looks much nicer. Changed.

What is the difference between payments and the result of the cart.billing.filter function?

Difference is, that between line 1115 and 1150 additional transaction records are pushed to the payments array.